volttron/utils/rmq_setup.py

```diff
@@ -183,62 +224,94 @@ def _create_federation_setup(admin_user, admin_password, is_ssl, vhost, vhome):
     federation_config_file = os.path.join(vhome,
                                           'rabbitmq_federation_config.yml')
     federation_config = read_config_file(federation_config_file)
-    federation = federation_config.get('federation-upstream')
+    federation = federation_config.get('federation-upstream', {})
 
-    if federation:
-        #ssl_params = None
-        #if is_ssl:
-            # ssl_params = rmq_mgmt.get_ssl_url_params()
+    for host, upstream in federation.items():
+        try:
+            name = "upstream-{host}-{vhost}".format(vhost=upstream['virtual-host'],
+                                                    host=host)
+            _log.debug("Upstream Server: {name} ".format(name=name))
+            if 'certificates' not in upstream:
+                _log.error(f"ERROR: Certificates key missing in config.\nContinuing with other configurations")
+                continue
+
+            rmq_user = upstream['federation-user']
+            # Build destination address
+            address = rmq_mgmt.build_remote_plugin_connection(rmq_user,
+                                                              host,
+                                                              upstream['port'],
+                                                              upstream['virtual-host'],
+                                                              is_ssl,
+                                                              certs_dict=federation[host]['certificates'])
+            prop = dict(vhost=vhost,
+                        component="federation-upstream",
+                        name=name,
+                        value={"uri": address})
+            rmq_mgmt.set_parameter('federation-upstream',
+                                   name,
+                                   prop,
+                                   vhost)
+
+            policy_name = 'volttron-federation'
+            policy_value = {"pattern": "^volttron",
+                            "definition": {"federation-upstream-set": "all"},
+                            "priority": 0,
+                            "apply-to": "exchanges"}
+            rmq_mgmt.set_policy(policy_name,
+                                policy_value,
+                                vhost)
```

Comment:
If setup was successful display message that it was successful. Else the last message before the command exits is "waiting for csr to be approved"